Assets/Scripts/UI/UIManager.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Serialization;
+using UnityEngine.UI;
+
+namespace UI
+{
+    public class UIManager : MonoBehaviour
+    {
+        [SerializeField] private DrunkMeter drunkMeter;
+        [SerializeField] private Multiplier multiplier;
+        [SerializeField] private TMP_Text score;
+        [SerializeField] private TMP_Text hp;
+        [SerializeField] private GameObject effectMeterPrefab;
+        [SerializeField] private GameObject effectMeterList;
+        
+        private Dictionary<string, EffectMeter> _effects = new Dictionary<string, EffectMeter>();
+        private const float PosOffset = 50.0f;
+        void Start()
+        {
+            RefreshUI();
+        }
+
+        private void Update()
+        {
+            UpdateEffects();
+        }
+
+        public void RefreshUI()
+        {
+            drunkMeter.SetDrunkValue();
+            multiplier.SetMultiplierValue();
+            score.text = Indestructibles.PlayerData.CurrentScore.ToString();
+            hp.text = Indestructibles.PlayerData.HealthPoints.ToString();
+        }
+
+        public void AddEffect(string effectName, Color effectColor, float effectDuration)
+        {
+            if (_effects.ContainsKey(effectName))
```

Comment:
Just to make sure: this is just the UI of the effects, right? The effect handling itself is still in the drinkables?

Reply:
yes, this only affects the sliders on the screen